feat: [a11y docn] Calendar: Update the accessibility documentation (#4856)
BrightspaceUI/core: components/calendar/README.md

| `max-value` | String | Maximum valid date that could be selected by a user. |
| `min-value` | String |  Minimum valid date that could be selected by a user. |
| `selected-value` | String | Currently selected date. |
| `summary` | String | ACCESSIBILITY: Summary of the calendar used by screen reader users for identifying the calendar and/or summarizing its purpose. |

### Events

* `d2l-calendar-selected`: dispatched when a date is selected through click, space, or enter. `e.detail.date` is in ISO 8601 calendar date format (`YYYY-MM-DD`).
<!-- docs: end hidden content -->

> ## Accessibility

Author: With calendar there was work done that deviated from the w3c recommendation, which is a bit hard to word in the more matter-of-fact bullet point style we've been using for these sections. Let me know suggestions on if we want that to be more consistent.

Reviewer: Should we capitalize `w3c`?

Reviewer 2: Yeah, this does deviate from the matter-of-fact style we use (and not just in bullets) in that it seems to try to explain how the project went, and even calls it the "date picker initiative" which some readers won't have context for (esp. if/when we take the site public) and even a lesson-learned "The calendar implementation emphasized the importance of thorough screen reader testing across all supported options."

I think we need to frame it more simply, without the project context.

> The Daylight calendar was built as a part of the date picker initiative and generally follows the w3c's best practice recommendations for a [Date picker dialog](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/). Of note is the keyboard behaviour following the [grid pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/).

Reviewer 3: ```suggestion
`d2l-calendar` was built as part of the date picker initiative and generally follows W3C's best practice recommendations for a [Date picker dialog](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/) with keyboard behaviour that follows the [grid pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/)
```

> The w3c recommendations, specifically relating to `attributes` on the calendar's table elements, were followed as much as possible. However, at the time of development it was not possible to create a calendar with the role of `grid` which had the desired appearance, functionality, and supported all browser/screen reader combinations that we support. We did aim to create the same experience for screen reader users as recommended by the example (e.g., announcing the expected content), but had to go about it in a more complex manner. The calendar implementation emphasized the importance of thorough screen reader testing across all supported options.

Reviewer: Just a couple small tweaks. Feel free to take or leave.
```suggestion
The W3C recommendations, specifically relating to `attributes` on the calendar's table elements, were followed as much as possible. However, at the time of development it was not possible to create a calendar with a `grid` role which had the desired appearance, functionality, and supported all browser/screen reader combinations that we support. We aimed to create the same experience for screen reader users as recommended by the example (e.g., announcing the expected content), but had to go about it in a more complex manner. The calendar implementation emphasized the importance of thorough screen reader testing across all supported options.
```
